apps/etterlatte-beregning/src/main/kotlin/avkorting/Avkorting.kt

```diff
@@ -270,9 +272,23 @@ data class Avkorting(
         return avkortetYtelseMedAllForventetInntekt
     }
 
-    fun hentAarsoppgjoer(fom: YearMonth): Aarsoppgjoer {
-        val funnet = aarsoppgjoer.find { it.aar == fom.year }
-        return funnet ?: Aarsoppgjoer(UUID.randomUUID(), aar = fom.year)
+    /*
+     * Hvilket årsoppgjør som er relevant basers på virkningstidspunkt.
+     * Hvis det ikke finnes et fra før på virkningstidspunkt opprettes et nytt.
+     *
+     * Ved innvilgelse/førstegangsbehandling så skal måneder før virkningsitdspunkt trekkes i fra
+     * forventa innvilgede måneder.
+     */
+    private fun hentAarsoppgjoer(
+        virkningstidspunkt: YearMonth,
+        innvilgelse: Boolean,
+    ): Aarsoppgjoer {
+        val funnet = aarsoppgjoer.find { it.aar == virkningstidspunkt.year }
+        return funnet ?: Aarsoppgjoer(
+            id = UUID.randomUUID(),
+            aar = virkningstidspunkt.year,
+            forventaInnvilgaMaaneder = if (innvilgelse) (12 - virkningstidspunkt.monthValue + 1) else 12,
+        )
```

Comment:
Hvis første virk i en behandling skal flyttes så vil ikke dette fungere. Men det er kanskje litt utenfor scope for denne oppgaven uansett. Jeg ville bare løfte fram behovet :p

Reply:
Jepp. Fikser i egen oppgave.